Sources/SQLite/Database/SQLiteConnection.swift

```diff
@@ -97,16 +98,22 @@ public final class SQLiteConnection: BasicWorker, DatabaseConnection, DatabaseQu
                         }
                     }
                 }
-                return promise.succeed(result: ())
+                self.eventLoop.execute {
+                    promise.succeed()
```

Comment:
succeeding / failing a promise is thread safe. we don't need to explicitly go on the event loop first.